variable/test/comparison_test.cpp

```diff
@@ -70,3 +70,130 @@ TEST(ComparisonTest, tolerance_type_mismatch) {
   EXPECT_THROW(is_approx(a, b, 0 /*implicit int*/), except::TypeError);
   EXPECT_NO_THROW(is_approx(a, b, 0.0f));
 }
+
+TEST(ComparisonTest, less_variances_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0},
+                                     Variances{1.0, 1.0});
+  const auto b = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0},
+                                     Variances{1.0, 1.0});
+  EXPECT_THROW(is_less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_dtypes_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  const auto b = makeVariable<int>(Dims{Dim::X}, Shape{2}, Values{0, 1});
+  EXPECT_THROW(is_less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_units_test) {
+  const auto a = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  auto b = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0});
+  b.setUnit(units::m);
+  EXPECT_THROW(is_less(a, b), std::runtime_error);
+}
```

Comment:
These things are (or should be) tested for the kernels you defined in `core::element`, and I think there is no need to re-test this here (same for more tests below).

Maybe a bit of history for the reasoning behind this: We have previously felt that our tests for operations for variables have become unmaintainable due to a large number of potential tests for all operations. We then figured out that it should be adequate to simple test `transform` as well as the operator passed to `transform`. The actual tests for the operation for variables could then be kept at a minimum (see also https://github.com/scipp/scipp/issues/513). Not all of the existing tests have been updated in this way, so this is definitely misleading.

If possible, we try to just write the "shortest possible" test that confirms that the correct operator is used (e.g., ensuring that `core::element::less` is called, and not `core::element::greate` due to copy and paste errors).

Reply:
I agree. I will modify the tests to only make one sanity check for one element variable.